runners/s3-benchrunner-rust/src/transfer_manager.rs

```diff
@@ -179,16 +212,44 @@ impl RunBenchmark for TransferManagerRunner {
         // We want the benchmark to fail fast if anything goes wrong,
         // so we're using a JoinSet.
         let mut task_set: JoinSet<Result<()>> = JoinSet::new();
-        for i in 0..self.config().workload.tasks.len() {
-            let task = self.clone().run_task(i);
-            task_set.spawn(task.instrument(tracing::Span::current()));
-        }
 
-        while let Some(join_result) = task_set.join_next().await {
-            let task_result = join_result.unwrap();
-            task_result?;
+        let workload_config = &self.config().workload;
+
+        if workload_config.checksum.is_some() {
+            return Err(SkipBenchmarkError("checksums not yet implemented".to_string()).into());
         }
+        if self.handle.transfer_path != None {
```

Comment:
utterly trivial: use `match` statement to unwrap the `transfer_path`
```suggestion
        match self.handle.transfer_path {
            Some(transfer_path) => {
                ... and you can use the unwrapped `transfer_path` variable in here
```